test/conditionals_test.dart

```diff
@@ -396,4 +411,21 @@ void main() {
       expect(e.runtimeType, equals(NonSupportedTypeException));
     }
   });
+
+  test(
+      'should return SignalRedrivenException when driven with '
+      'x signals and valid signals.', () async {
+    final mod = SignalRedrivenSequentialModuleWithX(Logic(), Logic(), Logic());
+    await mod.build();
+    final vectors = [
+      Vector({'a': 1, 'd': 1, 'c': 1}, {'b': LogicValue.z}),
```

Comment:
This would only be `z` the first cycle, since this is a `Sequential` in the test.

I think this case should hit the re-driven exception if `a` and `d` are both 1 for more than 1 cycle and `c` is valid (not `x` or `z`)